src/lib/modernExtend.ts

```diff
@@ -25,12 +29,26 @@ function getEndpointsWithInputCluster(device: Zh.Device, cluster: string | numbe
     return endpoints;
 }
 
+function getEndpointsWithOutputCluster(device: Zh.Device, cluster: string | number) {
```

Comment:
A lot of the code is the same as `getEndpointsWithInputCluster`, lets merge them into `getEndpointsWithCluster` and add `type: 'input' | 'output'` as an argument of it.

Reply:
Makes sense